src/GitLabHealth-Model-Importer/GitlabModelImporter.class.st

```diff
@@ -1,1569 +1,1642 @@
+
+	aAPIFile type = 'tree' ifTrue: [ 
+		^ GLHFileDirectory new
+			  name: aAPIFile name;
+			  yourself ].
+	^ GLHFileBlob new
+		  name: aAPIFile name;
+		  yourself
+]
+
+{ #category : #private }
+GitlabModelImporter >> detectEntityType: aType overAttribut: aSelector equalTo: value [
+
+	^ (self glhModel allWithType: aType) detect: [ :entity |
+		  (entity perform: aSelector) = value ] ifNone: [ nil ]. 
+]
+
+{ #category : #accessing }
+GitlabModelImporter >> glhApi [
+
+	self
+		deprecated: 'Use #repoApi instead'
+		on: '7 October 2024'
+		in:
+		'Pharo-11.0.0+build.726.sha.aece1b5473acf3830a0e082c1bc3a15d4ff3522b (64 Bit)'.
+
+	^ repoApi
+]
+
+{ #category : #accessing }
+GitlabModelImporter >> glhApi: anObject [
+
+	self
+		deprecated: 'Use #repoApi: instead'
+		on: '7 October 2024'
+		in:
+		'Pharo-11.0.0+build.726.sha.aece1b5473acf3830a0e082c1bc3a15d4ff3522b (64 Bit)'.
+
+	repoApi := anObject
+]
+
+{ #category : #accessing }
+GitlabModelImporter >> glhModel [
+
+	^ glhModel
+]
+
+{ #category : #accessing }
+GitlabModelImporter >> glhModel: anObject [
+
+	glhModel := anObject
+]
+
+{ #category : #'import - users' }
+GitlabModelImporter >> importActiveHumanUsers [
+
+	| params result users |
+	params := { 
+		#humans -> 'true'.
+		#active -> 'true'.
+		#without_project_bots -> 'true'
+	} asDictionary.
+	result := self repoApi users allWithParams: params.
+	users := (result collect: [ :usersJson | self parseUsersResult: usersJson ]) flattened.
+	
+	self glhModel
+				 addAll: users
+				 unless: self blockOnIdEquality.
+
+	^ users
+]
+
+{ #category : #'import - groups' }
+GitlabModelImporter >> importAllGroups [
+
+	| params results groups |
+	
+	params := { 
+			#top_level_only -> 'true'
+	} asDictionary.
+	results := self repoApi groups getAllWithParams: params.
+	
+	groups := (results collect: [ :groupsJson | generalReader
+			                    on: groupsJson readStream;
+			                    nextAs: #ArrayOfGroups. ]) flattened.
+	^ groups
+]
+
+{ #category : #'import - commits' }
+GitlabModelImporter >> importAndLoadLatestsCommitsOfProject: aGLHProject [
+
+	| commits completedProject |
+	completedProject := self completeImportProject: aGLHProject.
+	commits := self importLastestCommitsOfProject: completedProject.
+	commits do: [ :commit | self completeImportedCommit: commit ].
+	self chainsCommitsFrom: commits.
+	^ commits
+]
+
+{ #category : #'import-gitblame' }
+GitlabModelImporter >> importBlame: aGLHFile [ 
+	| results  params   jsonObject  result |
+			params := { 
+		    #ref ->   aGLHFile branch name  .
+	       } asDictionary.
+	           results := self repoApi blames getBlameFile: aGLHFile path ofProject: aGLHFile branch repository project id withParams: params      .
+              jsonObject := NeoJSONReader fromString: results .
+              result:= (jsonObject isArray ifTrue: [jsonObject ] ifFalse: [nil]).
+"              result:= result reject: [:item | item isNil]."
+   ^result 

```

Comment:
Blame are not part of the GitModel (yet) thus importing them to only return a dictionnary is not proper use of the Importer. 
Consider using a extension method outside of the importer (different project)